stock_owner_restriction/models/stock_move.py

```diff
@@ -1,14 +1,32 @@
 # Copyright 2020 Carlos Dauden - Tecnativa
 # Copyright 2020 Sergio Teruel - Tecnativa
+# Copyright 2023-2024 Quartile
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 from collections import defaultdict
 
-from odoo import models
+from odoo import api, fields, models
 
 
 class StockMove(models.Model):
     _inherit = "stock.move"
 
+    restrict_partner_id = fields.Many2one(
+        # Add compute to override the standard 'restrict_partner_id' behavior:
+        # assign the owner only when the picking type's restriction is 'picking_partner',
+        # and clear it for other restriction types to avoid unintended ownership.
```

Comment:
I think this should do. No need to explain the content of the compute method.
```suggestion
    # Add compute method to the standard field
    restrict_partner_id = fields.Many2one(
```